### PR TITLE
MULE-17994: Allow policy dispose to finish inflight events

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/internal/util/rx/TransactionAwareFluxSinkSupplier.java [hunk 1]
```diff
@@ -36,6 +42,10 @@ public TransactionAwareFluxSinkSupplier(Supplier<FluxSink<T>> sinkFactory, FluxS
 
   @Override
   public FluxSink<T> get() {
+    // if (disposed) {
```
Comment (balbifm) [inline core/src/main/java/org/mule/runtime/core/internal/util/rx/TransactionAwareFluxSinkSupplier.java]: Commented code

File: core/src/main/java/org/mule/runtime/core/internal/policy/DefaultPolicyManager.java [hunk 2]
```diff
@@ -92,27 +105,31 @@ public static boolean isNoPolicyOperation(OperationPolicy policy) {
...
   private Registry registry;
 
+  private CompositePolicyFactory compositePolicyFactory = new CompositePolicyFactory();
+
   private final AtomicBoolean isPoliciesAvailable = new AtomicBoolean(false);
 
-  private final Executor syncExecutor = Runnable::run;
+  private final ReferenceQueue<DeferredDisposable> stalePoliciesQueue = new ReferenceQueue<>();
+
+  private final Set<WeakReference<DeferredDisposable>> activePolicies = new HashSet<>();
```
[the remark this answers is not in the document]
Comment (elrodro83) [reply core/src/main/java/org/mule/runtime/core/internal/policy/DefaultPolicyManager.java]: added comment

**Comment by elrodro83** [comment]
--validate

**Comment by elrodro83** [comment]
--validate

**Comment by elrodro83** [comment]
--validate

**Pull request merged**